relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,230 @@
+        match message {
+            GlobalConfigManager::Get(sender) => {
+                sender.send(self.sender.borrow().clone());
+            }
+            GlobalConfigManager::Subscribe(sender) => {
+                sender.send(self.sender.subscribe());
+            }
+        }
+    }
+
+    /// Schedules the next global config request.
+    fn schedule_fetch(&mut self) {
+        if self.fetch_handle.is_idle() {
+            // XXX(iker): set a backoff mechanism?
+            self.fetch_handle
+                .set(self.config.global_config_fetch_interval());
+        }
+    }
+
+    /// Requests a new global config from upstream.
+    ///
+    /// As the upstream service is responsible for authentication, the global
+    /// config service doesn't deal with it.
+    fn update_global_config(&mut self) {
+        self.fetch_handle.reset();
+
+        if !self.config.has_credentials() {
+            return;
+        }
+
+        let upstream_relay: Addr<UpstreamRelay> = self.upstream.clone();
+        let internal_tx = self.internal_tx.clone();
+
+        tokio::spawn(async move {
+            let query = GetGlobalConfig::query();
+            let res = upstream_relay.send(SendQuery(query)).await;
+            if internal_tx.send(res).is_err() {
+                relay_log::error!("failed to forward internally the upstream response");
+            }
+        });
+    }
+
+    /// Handles the global config response from upstream, forwarding the global
+    /// config through the internal channel if succesfully received.
+    fn handle_upstream_response(&mut self, response: UpstreamResult) {
+        match response {
+            Ok(Ok(config)) => match config.global {
+                Some(global_config) => {
+                    if self.sender.send(Arc::new(global_config)).is_err() {
+                        relay_log::error!("failed to forward the global config internally");
+                    }
+                }
+                None => relay_log::error!("global config missing in upstream response"),
+            },
+            Ok(Err(e)) => relay_log::error!(
+                error = &e as &dyn std::error::Error,
+                "failed to send request to upstream"
+            ),
+            Err(e) => relay_log::error!(
+                error = &e as &dyn std::error::Error,
+                "failed to fetch global config from upstream"
+            ),
+        };
+        self.schedule_fetch();
+    }
+}
+
+impl Service for GlobalConfigService {
+    type Interface = GlobalConfigManager;
+
+    fn spawn_handler(mut self, mut rx: relay_system::Receiver<Self::Interface>) {
+        tokio::spawn(async move {
+            relay_log::info!("global config service starting");
+
+            // NOTE(iker): if this first request fails it's possible the default
+            // global config is forwarded. This is not ideal, but we accept it
+            // for now.
+            self.update_global_config();
```

Comment:
if we initiated the sleep_handle with 0 sec duration, then i think this line wouldn't be necessary, as it would request within the loop immediatly, what do you think?